### ISSUE TITLE
pd-server panic when collecting store metrics

### ISSUE THREAD

**Issue by lerencao** [pingcap/pd#1183]
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
pd panic with:

```
fatal error: concurrent map iteration and map write

goroutine 15792 [running]:
runtime.throw(0xfa9b43, 0x26)
        /data/program/tistack/tistack-release/local/go/src/runtime/panic.go:619 +0x81 fp=0xc423ee79f8 sp=0xc423ee79d8 pc=0x42b3f1
runtime.mapiternext(0xc423ee7aa8)
        /data/program/tistack/tistack-release/local/go/src/runtime/hashmap.go:747 +0x55c fp=0xc423ee7a88 sp=0xc423ee79f8 pc=0x40918c
github.com/pingcap/pd/server/core.(*StoresInfo).TotalWrittenBytes(0xc421db61c8, 0xc424cf3570)
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/core/store.go:430 +0x8f fp=0xc423ee7b18 sp=0xc423ee7a88 pc=0x8868cf
github.com/pingcap/pd/server/schedule.calculateWriteHotThreshold(0xc421db61c8, 0x3ff0000000000000)
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/schedule/hot_cache.go:104 +0x2f fp=0xc423ee7b38 sp=0xc423ee7b18 pc=0x89090f
github.com/pingcap/pd/server/schedule.(*HotSpotCache).CollectMetrics(0xc424d99300, 0xc421db61c8)
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/schedule/hot_cache.go:210 +0x185 fp=0xc423ee7c10 sp=0xc423ee7b38 pc=0x891205
github.com/pingcap/pd/server.(*coordinator).collectHotSpotMetrics(0xc42480a500)
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/coordinator.go:369 +0xc11 fp=0xc423ee7e80 sp=0xc423ee7c10 pc=0xc83961
github.com/pingcap/pd/server.(*RaftCluster).collectMetrics(0xc425006b60)
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/cluster.go:484 +0x121 fp=0xc423ee7f08 sp=0xc423ee7e80 pc=0xc7c531
github.com/pingcap/pd/server.(*RaftCluster).runBackgroundJobs(0xc425006b60, 0xdf8475800)
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/cluster.go:519 +0xf7 fp=0xc423ee7fd0 sp=0xc423ee7f08 pc=0xc7c907
runtime.goexit()
        /data/program/tistack/tistack-release/local/go/src/runtime/asm_amd64.s:2361 +0x1 fp=0xc423ee7fd8 sp=0xc423ee7fd0 pc=0x45aa01
created by github.com/pingcap/pd/server.(*RaftCluster).start
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/cluster.go:113 +0x528
```

2. What did you expect to see?

3. What did you see instead?

4. What version of PD are you using (`pd-server -V`)?
v2.0.2

**Comment by nolouch** [comment]
@lerencao thanks! 
This bug already fixes in v2.0.3, you can check the [CHANGELOG](https://github.com/pingcap/pd/blob/master/CHANGELOG.md#bug-fixes-1).

**Comment by lerencao** [comment]
@nolouch Great! Thanks, closing it.
